### ISSUE TITLE
Error when running query tasks regarding `LogseqMarkdownParser`

### ISSUE THREAD

**Issue by Positron010** [thiswillbeyourgithub/wdoc#5]
Hey,
Whenever I run a `query` task, I get this error to install `LogseqMarkdownParser` despite having it installed:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Scripts\wdoc.exe\__main__.py", line 7, in <module>
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\__init__.py", line 163, in cli_launcher
    _ = WDoc(**kwargs)
        ^^^^^^^^^^^^^^
  File "<@beartype(WDoc.WDoc.WDoc.__init__) at 0x1c5ab109b20>", line 14, in __init__
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\utils\misc.py", line 679, in new_func
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\WDoc.py", line 545, in __init__
    self.query_task(query=query)
  File "<@beartype(WDoc.WDoc.WDoc.query_task) at 0x1c5ab1345e0>", line 36, in query_task
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\WDoc.py", line 1611, in query_task
    output = rag_chain.invoke(
             ^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\runnables\base.py", line 3020, in invoke
    input = context.run(step.invoke, input, config, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\runnables\base.py", line 4706, in invoke
    return self._call_with_config(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\runnables\base.py", line 1923, in _call_with_config
    context.run(
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\runnables\config.py", line 396, in call_func_with_variable_args
    return func(input, **kwargs)  # type: ignore[call-arg]
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\runnables\base.py", line 4562, in _invoke
    output = call_func_with_variable_args(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\runnables\config.py", line 396, in call_func_with_variable_args
    return func(input, **kwargs)  # type: ignore[call-arg]
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\WDoc\WDoc.py", line 1532, in retrieve_documents
    "unfiltered_docs": retriever.get_relevant_documents(inputs["question_for_embedding"]),
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\_api\deprecation.py", line 176, in warning_emitting_wrapper
    if not warned and not is_caller_internal():
                          ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\langchain_core\_api\internal.py", line 17, in is_caller_internal
    caller_module = inspect.getmodule(frame)
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\inspect.py", line 1000, in getmodule
    if ismodule(module) and hasattr(module, '__file__'):
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\lazy_import\__init__.py", line 156, in __getattribute__
    _load_module(self)
  File "C:\Users\Habib\AppData\Local\Programs\Python\Python312\Lib\site-packages\lazy_import\__init__.py", line 536, in _load_module
    raise_from(ImportError(
  File "<string>", line 3, in raise_from
ImportError: WDoc attempted to use a functionality that requires module LogseqMarkdownParser, but it couldn't be loaded. Please install LogseqMarkdownParser and retry.

```

While running `summarize` tasks is successful, I also have this error:

```
Error when unlazyloading module 'LogseqMarkdownParser'. Error: 'WDoc attempted to use a functionality that requires module LogseqMarkdownParser, but it couldn't be loaded. Please install LogseqMarkdownParser and retry.'
This can be caused by beartype's typechecking
```

Thanks

**Comment by thiswillbeyourgithub** [comment]
Hi, thanks for giving wdoc a try.

Can you try, in this order, each of the potential fix, then try wdoc, and if it fails move to the next fix?

1. pip install -U wdoc
2. pip install -U logseqmarkdownparser
3. pip install -U beartype
4. call python with `WDOC_TYPECHECKING='disabled'`

**Comment by Positron010** [comment]
Hi,
I just tried all the commands in the provided order, and still got the same error (all of the first three commands provided that all requirements were satisfied). As for the last command, I am not sure if I've done it right as I just made an environment variable with the provided name and value; not sure if this is what you mean.

Thanks

**Comment by thiswillbeyourgithub** [comment]
4. Call python with  `WDOC_TYPECHECKING='disabled' python [youargs]` or ` WDOC_TYPECHECKING='disabled' wdoc [yourargs]`
5. ~~Try the latest version I just pushed in this repo at the dev branch.~~ not yet


Also thanks a lot for taking the time to create an issue, that really helps!

**Comment by Positron010** [comment]
Sorry for the late reply. I had to figure out how to execute the last command on PowerShell, because I'm using Windows. I just ran the command and got the same error above.

**Comment by thiswillbeyourgithub** [comment]
Claude Sonnet tells me that the proper syntax for powershell to set an environment variable is `$env:WDOC_TYPECHECKING = "disabled"`

**Comment by Positron010** [comment]
Yes, I tried that exact command.

**Comment by thiswillbeyourgithub** [comment]
Can you add --verbose and see if something comes up ? I'm surprised disabling typechecking does not fix this.

**Comment by Positron010** [comment]
In addition to the above error. I also get (with `--verbose`):

```
Couldn't import optional package 'ftlangdetect', trying to import langdetect (but it's much slower): 'No module named 'ftlangdetect''

```

and

```
Failed to import optional package 'pdftotext': 'No module named 'pdftotext''
Not using a pdf loader timeout
```

**Comment by thiswillbeyourgithub** [comment]
Yeah those are know quircks of windows. You can try pip install wdoc[fasttext] and wdoc[pdftotext] if you'd like but that's not related

**Comment by thiswillbeyourgithub** [comment]
Just pushed 1.9.3 for you. It's in the dev branch

Try it, and if it fails try with the env variable WDOC_DISABLE_LAZYLOADING="true"

**Comment by Positron010** [comment]
I get this:
```
AssertionError: Unexpected key for WDOC env variable: WDOC_DISABLE_LAZYLOADING
```

**Comment by thiswillbeyourgithub** [comment]
Are you sure you're using the dev branch? can you run wdoc --version ? It should be 1.9.3

**Comment by thiswillbeyourgithub** [comment]
Actually I Just pushed 1.9.4 because I rebased the 1.9.3 too

**Comment by Positron010** [comment]
Yes, I was using version 1.9.3
The 1.9.4 version I just installed does not print its version. I get this error:
```
ModuleNotFoundError: No module named 'LogseqMarkdownParser'
```

**Comment by thiswillbeyourgithub** [comment]
Weird. Just pushed 1.9.5

**Comment by thiswillbeyourgithub** [comment]
(I'm the dev behing LogseqMarkdownParser too so it's extra annoying because there's no reason for this specific package to have issues)

**Comment by Positron010** [comment]
Thanks for your efforts. I still get the same `ModuleNotFoundError` error with the latest version. I am afraid I shall leave and catch up regarding this later on. Thanks again.

**Comment by thiswillbeyourgithub** [comment]
Alright. Sorry that didn't work out and thanks for making me aware of this issue.

Next time try "pip install LogseqMarkdownParser" but that looks like a windows/python problem because it definitely is in the setup.py

**Comment by thiswillbeyourgithub** [comment]
Until you give it a try I'll close this as I can't reproduce on my end

**Comment by Positron010** [comment]
Hi again,
So I think I found out the issue. `pip` was not installing the `LogseqMarkdownParser` successfully; it was just making the `.dist-info` folder. See [here](https://stackoverflow.com/questions/50585246/pip-install-creates-only-the-dist-info-not-the-package) for more info.

I resolved the issue by editing the `setup.py` file in `LogseqMarkdownParser` removing the `package_dir={"": "src"},` line, and moving the `src/LogseqMarkdownParser` folder to the root folder, and then `pip install .`. Not sure if this is the optimal way, but python detected the package after that.
